### PR TITLE
Pin merging service

### LINKED ISSUE #168: Add API for merging issues

### PR DESCRIPTION
**Pull request by supasate**
Close #168 and #172 

**How to use**

`POST /pins/:pinId/merging -d { mergedParentPin: <parentPinId> }`

### CODE REVIEW

File: src/services/pin-merging/hooks/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const auth = require('feathers-authentication').hooks;
+
+const logActivity = require('./log-activity');
+const prepareActivityLog = require('./prepare-activity-log');
+const ORGANIZATION_ADMIN = require('../../../constants/roles').ORGANIZATION_ADMIN;
```
Comment (parnurzeal) [inline src/services/pin-merging/hooks/index.js:6]: nit: newline above separating hooks and constants.

File: src/services/pin-merging/hooks/prepare-activity-log.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+const errors = require('feathers-errors');
+const mongoose = require('mongoose');
+
+const MERGE_PIN = require('../../../constants/actions').MERGE_PIN;
+const MERGING = require('../../../constants/actions').types.MERGING;
+const Pin = require('../../pin/pin-model');
```
Comment (parnurzeal) [inline src/services/pin-merging/hooks/prepare-activity-log.js]: Model before fixtures + constants

File: test/services/pin-merging/hooks/prepare-activity-log.test.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+// Test helper functions
+const assertTestEnv = require('../../../test_helper').assertTestEnv;
+const expect = require('../../../test_helper').expect;
+const loadFixture = require('../../../test_helper').loadFixture;
+const stub = require('../../../test_helper').stub;
+
+// Models
+const PinModel = require('../../../../src/services/pin/pin-model');
```
Comment (parnurzeal) [inline test/services/pin-merging/hooks/prepare-activity-log.test.js]: Delete "Model" word.

File: test/services/pin-merging/hooks/prepare-activity-log.test.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+// Fixtures
+const pins = require('../../../fixtures/pins');
+const ASSIGNED_PIN_ID = require('../../../fixtures/constants').ASSIGNED_PIN_ID;
+const ASSIGNED_PIN_DETAIL = require('../../../fixtures/constants').ASSIGNED_PIN_DETAIL;
+const VERIFIED_PIN_ID = require('../../../fixtures/constants').VERIFIED_PIN_ID;
```
Comment (parnurzeal) [inline test/services/pin-merging/hooks/prepare-activity-log.test.js]: sorting

File: test/fixtures/pins.js [hunk 5]
```diff
@@ -1,12 +1,21 @@
-const casual = require('casual');
+const ObjectId = require('mongoose').Types.ObjectId;
+
+const ORGANIZATION_ID = require('./constants').ORGANIZATION_ID;
+const NORMAL_DEPARTMENT_ID = require('./constants').NORMAL_DEPARTMENT_ID;
+const ASSIGNED_PIN_ID = require('./constants').ASSIGNED_PIN_ID;
+const ASSIGNED_PIN_DETAIL = require('./constants').ASSIGNED_PIN_DETAIL;
```
Comment (parnurzeal) [inline test/fixtures/pins.js]: sorting

File: test/fixtures/constants.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const SUPER_ADMIN_USER_ID = '579334c74443625d6281b6ff';
+
+module.exports = {
+  ASSIGNED_PIN_ID,
+  ASSIGNED_PIN_DETAIL,
+  UNVERIFIED_PIN_ID,
+  UNVERIFIED_PIN_DETAIL,
+  VERIFIED_PIN_ID,
+  VERIFIED_PIN_DETAIL,
+  ORGANIZATION_ID,
```
Comment (parnurzeal) [inline test/fixtures/constants.js:22]: sorting

File: src/services/pin-merging/hooks/index.js (hunk 1 above)
Comment (supasate) [reply src/services/pin-merging/hooks/index.js:6]: ok

File: src/services/pin-merging/hooks/prepare-activity-log.js (hunk 2 above)
Comment (supasate) [reply src/services/pin-merging/hooks/prepare-activity-log.js]: ok

File: test/services/pin-merging/hooks/prepare-activity-log.test.js (hunk 3 above)
Comment (supasate) [reply test/services/pin-merging/hooks/prepare-activity-log.test.js]: ok

File: test/services/pin-merging/hooks/prepare-activity-log.test.js (hunk 4 above)
Comment (supasate) [reply test/services/pin-merging/hooks/prepare-activity-log.test.js]: ok. Renamed to PIN_ASSIGNED_ID, PIN_ASSIGNED_DETAIL, DEPARTMENT_GENERAL_ID, etc.

File: test/fixtures/pins.js (hunk 5 above)
Comment (supasate) [reply test/fixtures/pins.js]: ok

File: test/fixtures/constants.js (hunk 6 above)
Comment (supasate) [reply test/fixtures/constants.js:22]: ok

**Comment by supasate** [comment]
Done

**Pull request merged**